prov/util/src/util_mr_cache.c

```diff
@@ -150,6 +150,25 @@ void ofi_mr_cache_notify(struct ofi_mr_cache *cache, const void *addr, size_t le
 		util_mr_uncache_entry(cache, entry);
 }
 
+void ofi_mr_cache_flush_unmapped(struct ofi_mr_cache *cache)
+{
+	struct ofi_mr_entry *entry;
+
+	pthread_mutex_lock(&cache->monitor->lock);
+	while (!dlist_empty(&cache->flush_list)) {
+		dlist_pop_front(&cache->flush_list, struct ofi_mr_entry,
+				entry, list_entry);
+		FI_DBG(cache->domain->prov, FI_LOG_MR, "flush %p (len: %zu)\n",
+		       entry->info.iov.iov_base, entry->info.iov.iov_len);
+		pthread_mutex_unlock(&cache->monitor->lock);
+
+		util_mr_free_entry(cache, entry);
+		pthread_mutex_lock(&cache->monitor->lock);
+	}
+
+	pthread_mutex_unlock(&cache->monitor->lock);
+}
```

Comment:
The commit message doesn't provide any real motivation for this change, other than to say it's different.  Can you provide a stronger justification in the commit message, especially since there's nothing provided to call this?

Because this function duplicates the upper half of of ofi_mr_cache_flush, it may be better to adjust ofi_mr_cache_flush to skip LRU processeing based on an input parameter.

Reply:
Thank you for reviewing! I added an argument ```flush_lru``` to ofi_mr_cache_flush and updated comments.